src/parser/jobs/sam/Gauge.js

```diff
@@ -0,0 +1,208 @@
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+	//check for kenki value changes, then sen changes
+	_onCast(event) {
+		const abilityId = event.ability.guid
+		if (KENKI_BUILDERS[abilityId]) {
+			this._addKenki(abilityId)
+		}
+		if (KENKI_SPENDERS[abilityId]) {
+			this._kenki -= KENKI_SPENDERS[abilityId]
+		}
+		if (abilityId === ACTIONS.HAGAKURE.id) {
+			this._Sen2Kenki()
+		}
+		if (abilityId ===ACTIONS.HIGANBANA.id || abilityId === ACTIONS.TENKA_GOKEN.id || abilityId === ACTIONS.MIDARE_SETSUGEKKA.id) {
+			this._removeSen()
+		}
+		if (abilityId === ACTIONS.GEKKO.id || abilityId === ACTIONS.MANGETSU.id) {
+			this._addGekkoSen()
+		}
+		if (abilityId === ACTIONS.KASHA.id || abilityId === ACTIONS.OKA.id) {
+			this._addKashaSen()
+		}
+		if (abilityId === ACTIONS.YUKIKAZE.id) {
+			this._addYukikazeSen()
+		}
+
+	}
+
+	//kenki quick maths
+
+	_addKenki(abilityId) {
+		this._kenki += KENKI_BUILDERS[abilityId]
+		if (this._kenki > MAX_KENKI) {
+			const waste = this._kenki - MAX_KENKI
+			this._wastedKenki += waste
+			this._kenki = MAX_KENKI
+			return waste
+		}
+		return 0
+	}
+
+	_Sen2Kenki() {
+		this._kenki += ((this._gekkosen + this._kashasen + this._yukikazesen) * 20)
+		if (this.kenki > MAX_KENKI) {
+			const waste = this._kenki - MAX_KENKI
+			this._wastedKenki += waste
+			this._kenki = MAX_KENKI
+			return waste
+		}
+
+		this._gekkosen = 0
+		this._kashasen = 0
+		this._yukikazesen = 0
+
+		return 0
+	}
+
+	//sen calcs
+
+	_addGekkoSen() {
+		this._gekkosen += 1
+		if (this._gekkosen > MAX_GEKKO_SEN) {
+			const waste = this._gekkosen - MAX_GEKKO_SEN
+			this._wastedsen += waste
+			return waste
+		}
+		return 0
+	}
+
+	_addKashaSen() {
+		this._kashasen += 1
+		if (this._kashasen > MAX_KASHA_SEN) {
+			const waste = this._kashasen - MAX_KASHA_SEN
+			this._wastedsen += waste
+			return waste
+		}
+		return 0
+	}
+
+	_addYukikazeSen()  {
+		this._yukikazesen += 1
+		if (this._yukikazesen > MAX_YUKIKAZE_SEN) {
+			const waste = this._yukikazesen - MAX_YUKIKAZE_SEN
+			this._wastedsen += waste
+			return waste
+		}
+		return 0
+	}
+
+	_removeSen() {
+		this._gekkosen = 0
+		this._kashasen = 0
+		this._yukikazesen = 0
+
+		return 0
+	}
+
+	_onDeath() {
+		// Death just flat out resets everything. Stop dying.
+		this._wastedKenki += this._kenki
+		this._kenki = 0
+
+		this._wastedsen += (this._gekkosen + this._kashasen + this._yukikzaesen)
+		this._gekkosen = this._kashasen = this._yukikazesen = 0
+	}
+
+	_onComplete() {
+		//kenki suggestions
+		if (this._wastedKenki >= 20) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.HAKAZE.icon,
+				content: <Fragment>
+					You used kenki builders in a way that overcapped you.
+				</Fragment>,
+				severity: this._wastedKenki === 20? SEVERITY.MINOR : this._wastedKenki >= 50? SEVERITY.MEDIUM : SEVERITY.MAJOR,
```

Comment:
Could use `TieredSuggestions` for this instead.

Reply:
I'd say should, not just could. Or at the very least fix the bug that flags the 21-49 range as `MAJOR` and the 50+ range as `MEDIUM`. :blobmorning: